### PR TITLE
Implemented error catcher & 404 page

### LINKED ISSUE #181: Add Custom Error Pages and Error Handling
### Description 
We need to improve error handling by adding custom pages and catchers to provide a better user experience and align with the design guidelines.

### Tasks:

Create a custom 404 page that matches the design and provides navigation options for users.
Add a global Error page to handle unexpected errors with an option to retry actions.
Implement localized error catchers for specific routes where necessary.
Ensure all error pages are consistent with the overall design.
Set up error logging to help with debugging and tracking issues.

### PR DESCRIPTION
**Pull request by jedstroke**
## Detailed Information

image
*<small>404 page</small>*

image
*<small>Error logging UI</small>*

image
*<small>Error logger dashboard Sentry</small>*

## Related Issues

Closes #181

---

## Type of Change

- [x] 🐛 Bug fix or ⚙️ enhancement
- [x] ✨ New feature or Chore (change with no new features or fixes)
- [x] 📚 Documentation update

---

## Checklist 
- [x] The code follows the style guidelines of this project.
- [x] All new and existing tests pass.
- [x] This pull request is ready to be merged and reviewed.

### CODE REVIEW

**Comment by fishonamos** [comment]
Looks cool!

File: app/client/src/app/error.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+'use client'
+
+import * as Sentry from "@sentry/nextjs"
```
Comment (Solomonsolomonsolomon) [inline app/client/src/app/error.tsx:3]: no need to use sentry, just implement error handling using next 
read https://nextjs.org/docs/app/building-your-application/routing/error-handling for more details

Comment (jedstroke) [reply app/client/src/app/error.tsx:3]: I will just do this

**Comment by fishonamos** [comment]
Good!. You moved the doc files out of the docs folder. Can you revert them?

**Comment by jedstroke** [comment]
> Good!. You moved the doc files out of the docs folder. Can you revert them?

Hang on, I am doing it now

**Comment by fishonamos** [comment]
LGTM. @Solomonsolomonsolomon

File: app/client/src/lib/error-logger.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (Solomonsolomonsolomon) [inline app/client/src/lib/error-logger.ts:1]: sentry is still used here, take out sentry completely

**Review by Solomonsolomonsolomon — CHANGES_REQUESTED** [review]

File: app/client/src/lib/error-logger.ts (hunk 2 above)
Comment (jedstroke) [reply app/client/src/lib/error-logger.ts]: > sentry is still used here, take out sentry completely
Is that all buddy?

**Comment by Kingsuite** [comment]
This is my observation: 
https://www.loom.com/share/d4cba707ddf34537bfd7a74f76a8fb1c

**Pull request merged**